main.c

```diff
@@ -430,7 +430,12 @@ STATIC bool run_code_py(safe_mode_t safe_mode) {
             // it may also return due to another interrupt, that's why we check
             // for deep sleep alarms above. If it wasn't a deep sleep alarm,
             // then we'll idle here again.
-            port_idle_until_interrupt();
+        #if CIRCUITPY_ALARM
+	    alarm_pretending_deep_sleep();
+        #else
+	    port_idle_until_interrupt();
+        #endif
+	    return false; // to go REPL
```

Comment:
Can you explain this return? If you return here after a fake sleep alarm occurs, won't it skip the `alarm_woken_from_sleep` check that instructs it to soft reboot?

Reply:
@hierophect In earlier design of pretended deep sleep, it enters REPL first when an event occurs and then does soft reboot at Ctrl-D in REPL. (please look at the state diagram discussed before)
After the discussion with @dhalbert the design has changed to do soft reboot directly. So, return false is no longer needed.

Reply:
I see that the `reset_cpu()` in `alarm_pretending_deep_sleep` actually hard reboots the CPU when waking, circumventing the alarm detection process entirely. Unfortunately, this may actually be what is causing @dhalbert's problems with USB, because it will force re-enumeration. Since the idea of fake sleeping is to avoid any serial disconnection, we may need a different strategy here. @dhalbert will have more details.

Reply:
@jun2sak I missed the fact that you are hard-resetting the board. That is why the USB connection is going away for me, and maybe sometimes for you. On Windows, Tera Term recovers from this better. The CPU should not be hard-reset after a simulated deep sleep.

Reply:
@dhalbert, @hierophect I understand. I think `reset_cpu()` can be replaced with something like soft reboot. 
Then, I wonder again there would be little difference between deep sleep and light sleep.  Original deep sleep causes hard-reset and implies re-initialization of USB, however, in the new design, both of deep sleep and light sleep do soft waiting (idle loop) and soft reboot. Is that OK ?

Reply:
I think that is fine. The whole light-sleep/deep-sleep terminology was based on using some terms from the ESP32-S2 IDF (SDK). And it turned out that we couldn't use their light sleep anyway because it shut down too many things. We haven't really found many chips where there is a meaningful light sleep that we cannot already implement by doing a WFI instruction, for instance. I would like to rethink this a bit and perhaps change the terminology incompatibly in 7.0.0, but we do not have to do that now.

Reply:
I think the confusion here is that there are two different deep sleeps. One is "pretending" to deep sleep in order to iterate on deep sleep code when connected over USB. The other is real deep sleep when not connected to USB where everything does actually shutdown. These two modes look the same from the Python code but internally pretend deep sleep looks a lot like light sleep. So, I think deep sleep and light sleep *are* different and appropriate. Light sleep is pretty similar to `time.sleep` but allows for other ways to wake up. I think these are the right APIs.